### PR TITLE
Improvements to block volume encoding

### PR DESCRIPTION
**Pull request by Schnjan**
Following improvements were made:

* The dct coefficients are now calculated in a more generalized way, allowing different block sizes to be used.
* Added some simple tests for the dct.
* The quantization matrix used for compression is now stored in the block volume data stream and used for deserialization, which allows us to change the calculation algorithm of the quantization matrix later
* The way the quantization matrix is calculated is now generic and allows different block sizes to be used.
* Updated the volume library to .NET 8
* Introduced the GOM volume format as readable format, with readers for float, int16 and uint16 volumes
* Used SIMD instructions to speed up dct. This works only for block size N=8 though, but other block sizes don't seem to offer advantages

### CODE REVIEW

File: src/PiWeb.Volume.Compare/PiWeb.Volume.Compare.csproj [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net8.0-windows10.0.22621</TargetFramework>
```
Comment (georg-eckert-zeiss) [inline src/PiWeb.Volume.Compare/PiWeb.Volume.Compare.csproj:4]: can we do this without targeting windows? `<TargetFramework>net8.0</TargetFramework>`?

Comment (georg-eckert-zeiss) [reply src/PiWeb.Volume.Compare/PiWeb.Volume.Compare.csproj:4]: that way we can use a linux runner

File: src/PiWeb.Volume.Convert/ConvertVolume.cs [hunk 2]
```diff
@@ -76,26 +69,61 @@ public static Volume FromVgi(
...
+			double maxValue,
+			bool streamed,
+			IProgress<double>? progress = null,
+			ILogger? logger = null )
+		{
+			var sw = Stopwatch.StartNew();
+
+			try
+			{
+				var voxelStream = dataType switch
```
Comment (georg-eckert-zeiss) [inline src/PiWeb.Volume.Convert/ConvertVolume.cs:92]: might be easier to use a tuple here: 

```cs
(dataType, extraPolate) switch
{
  ( Gom.DataType.Int16, false ) => ...
}
```

File: src/PiWeb.Volume.Convert/FloatStream.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss IMT (IZfM Dresden)                   */
```
Comment (georg-eckert-zeiss) [inline src/PiWeb.Volume.Convert/FloatStream.cs:4]: use the new copyright

File: src/PiWeb.Volume.Convert/FloatStream.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2020                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Volume.Convert
```
Comment (georg-eckert-zeiss) [inline src/PiWeb.Volume.Convert/FloatStream.cs:11]: file-scoped

Comment (georg-eckert-zeiss) [reply src/PiWeb.Volume.Convert/FloatStream.cs:11]: might as well change the whole solution

**Review by georg-eckert-zeiss — CHANGES_REQUESTED** [review]
Looks good. What about the windows target?

Don't forget to give hints about semantic changes by adding (Minor) or (Major) to the commit messages.

File: src/PiWeb.Volume.Compare/PiWeb.Volume.Compare.csproj (hunk 1 above)
Comment (Schnjan) [reply src/PiWeb.Volume.Compare/PiWeb.Volume.Compare.csproj:4]: Unfortunately not, that's a WPF project.

File: src/PiWeb.Volume.Convert/ConvertVolume.cs (hunk 2 above)
Comment (Schnjan) [reply src/PiWeb.Volume.Convert/ConvertVolume.cs:92]: I don't think this improves the readability

**Comment by georg-eckert-zeiss** [comment]
All fine.

**Review by georg-eckert-zeiss — APPROVED** [review]

**Pull request merged**